ui/src/router/components/form/components/router_config/route_card/RouteCard.js

```diff
@@ -13,16 +13,23 @@ import { SelectEndpointComboBox } from "../../../../../../components/form/endpoi
 import { EuiFieldDuration } from "../../../../../../components/form/field_duration/EuiFieldDuration";
 import { get } from "../../../../../../components/form/utils";
 import "./RouteCard.scss";
+import { EuiComboBoxSelect } from "../../../../../../components/form/combo_box/EuiComboBoxSelect";
 
 export const RouteCard = ({
   route,
+  protocol,
   endpointOptions,
   onChange,
   onDelete,
   errors,
 }) => {
   const [endpointsMap, setEndpointsMap] = useState({});
```

Comment:
Eventually, I imagine we will filter the available endpoints by the chosen protocol either in `ui/src/providers/endpoints/MerlinEndpointsProvider.js` or here, right?

Reply:
For UPI yes, actually they should all be `"/caraml.upi.v1.UniversalPredictionService/PredictValues"`, only if there is custom GRPC then a free form would be allowed. 

should we fix the route service method and disable user modification? @pradithya do you foresee any issue?

Reply:
Yes, I am fine with that

Reply:
Actually my question is more about the HTTP vs UPI Merlin endpoints - we currently get all endpoints in the `MerlinEndpointsProvider`, that are active under the same project. We should filter only those endpoints that match the protocol.

> should we fix the route service method and disable user modification?

This sounds good to me too.

Reply:
right, added the filter to `ui/src/router/components/form/components/router_config/RoutesPanel.js`. Assuming all grpc endpoint will not start with `http` while http endpoints will always have `http`.